docs/frontend-system/architecture/07-routes.md

```diff
@@ -8,143 +8,420 @@ description: Frontend routes
 
-const link = useRouteRef(indexPageRouteRef)
+const createComponentRouteRef = createRouteRef();
 
-// scaffolder
+// plugins/scaffolder/src/plugin.tsx
+import { createPlugin, createPageExtension } from '@backstage/frontend-plugin-api';
+import { createComponentRouteRef } from './routes';
 
-export const catalogIndexPageRouteRef = createExternalRouteRef({
-  defaultTarget: 'catalog/index',
-})
+const createComponentPage = createPageExtension({
+  defaultPath: '/'
+  routeRef: createComponentRouteRef,
+  loader: async () => (
+    <div>
+      <h1>Create Component</h1>
+    </div>
+  )
+});
 
-export const scaffolderPlugin = createPlugin({
-  id: 'scaffolder,
-  externalRoutes: {
-    catalogIndex: catalogIndexPageRouteRef,
+export default const createPlugin({
+  id: 'scaffolder',
+  routes: {
+    createComponent: createComponentRouteRef,
   },
-})
+  extensions: [createComponentPage]
+});
+
+// plugins/scaffolder/src/index.ts
+export { default } from './plugin';
+```
+
+On important thing to highlight is that it is currently not possible to have parameterized `ExternalRouteRefs`, or to bind an external route to a parameterized route, although this may be added in the future if needed.
+
+Now let's move on and configure the app to point to the Scaffolder create component page when the catalog create component ref be used.
+
+### Binding External Route References
 
+The association of external routes is controlled by the app. Each `ExternalRouteRef` of a plugin should be bound to an actual `RouteRef`, usually from another plugin. The binding process happens once at app startup, and is then used through the lifetime of the app to help resolve concrete route paths.
 
-import {catalogIndexPageRouteRef} from '../../routes
+Using the above example of the Catalog entities list page to the Scaffolder create component page, we might do something like this in the app configuration file:
 
-const link = useRouteRef(catalogIndexPageRouteRef)
+```yaml
+# app-config.yaml
+app:
+  routes:
+    bindings:
+      plugin.catalog.externalRoutes.createComponent: plugin.scaffolfer.routes.createComponent
+```
 
-// app
+Or via code, in the file where the app is created:
 
-import {catalogPlugin} from '@backstage/plugin-catalog'
+```tsx
+// packages/app/src/App.tsx
+import { createApp } from '@backstage/frontend-app-api';
+import catalog from '@backstage/plugin-catalog';
+import scaffolder from '@backstage/plugin-scaffolder';
 
 const app = createApp({
-  bindRoutes({bind}) {
-    bind(scaffolderPlugin, {
-      catalogIndex: catalogPlugin.routes.index,
-    })
+  bindRoutes({ bind }) {
+    bind(catalog.externalRoutes, {
+      createComponent: scaffolder.routes.createComponent,
+    });
   },
-})
-*/
+});
+
+export default app.createRoot();
+
+// packages/app/src/index.ts
+import ReactDOM from 'react-dom/client';
+import app from './App';
+
+ReactDOM.createRoot(document.getElementById('root')!).render(app);
```

Comment:
Tbh this bit feels a bit unnecessary, maybe just leave it at `createApp`?

Reply:
Removed